src/SFA.DAS.AANHub.Application/Apprentices/Commands/PatchApprenticeMember/PatchApprenticeMemberCommandHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿using MediatR;
+using System.Text.Json;
+using SFA.DAS.AANHub.Application.Mediatr.Responses;
+using SFA.DAS.AANHub.Domain.Entities;
+using SFA.DAS.AANHub.Domain.Models;
+using SFA.DAS.AANHub.Domain.Interfaces.Repositories;
+using SFA.DAS.AANHub.Domain.Interfaces;
+using static SFA.DAS.AANHub.Domain.Common.Constants;
+
+namespace SFA.DAS.AANHub.Application.Apprentices.Commands.PatchApprenticeMember
+{
+    public class PatchApprenticeMemberCommandHandler : IRequestHandler<PatchApprenticeMemberCommand, ValidatedResponse<PatchApprenticeMemberCommandResponse>>
+    {
+        private readonly IApprenticesWriteRepository _apprenticesWriteRepository;
+        private readonly IAuditWriteRepository _auditWriteRepository;
+        private readonly IAanDataContext _aanDataContext;
+
+        public PatchApprenticeMemberCommandHandler(IApprenticesWriteRepository apprenticesWriteRepository,
+            IAuditWriteRepository auditWriteRepository,
+            IAanDataContext aanDataContext)
+        {
+            _apprenticesWriteRepository = apprenticesWriteRepository;
+            _auditWriteRepository = auditWriteRepository;
+            _aanDataContext = aanDataContext;
+        }
+
+        public async Task<ValidatedResponse<PatchApprenticeMemberCommandResponse>> Handle(PatchApprenticeMemberCommand command, CancellationToken cancellationToken)
+        {
+            var success = true;
+
+            var apprentice = await
+                _apprenticesWriteRepository.GetApprentice(command.ApprenticeId);
+
+            if (apprentice == null)
+                return new ValidatedResponse<PatchApprenticeMemberCommandResponse>((PatchApprenticeMemberCommandResponse)null!);
+
+            var audit = new Audit()
+            {
+                Action = "Patch",
+                ActionedBy = command.RequestedByMemberId ?? Guid.Empty,
+                AuditTime = DateTime.UtcNow,
+                Before = JsonSerializer.Serialize(apprentice),
+                Resource = MembershipUserType.Apprentice
+            };
+
+            var patchedApprentice = (ApprenticeMemberPatchModel)apprentice;
+
+            command.Patchdoc.ApplyTo(patchedApprentice);
+
+            apprentice.Email = patchedApprentice.Email;
+            apprentice.Name = patchedApprentice.Name;
```

Comment:
What if, we remove line#46 and apply patch directly to the retrieved apprentice `command.Patchdoc.ApplyTo(apprentice)` would that work? If it would, we can also remove line 50 and 51, and also get rid of ApprenticeMemberPatchModel.

Reply:
ApprenticeMemberPatchModel is no longer used `public JsonPatchDocument<ApprenticeMemberPatchModel> Patchdoc { get; set; } = null!;` is now `public JsonPatchDocument<Apprentice> Patchdoc { get; set; } = null!;`